### PR TITLE
(POOLER-192) Use Rubocop 1.0

### PR DESCRIPTION
**Pull request by rooneyshuman**
Fixes errors resulting from using Rubocop's new stable version v1.0 and updates the gem.

### CODE REVIEW

File: lib/vmpooler/pool_manager.rb [hunk 1]
```diff
@@ -867,16 +868,12 @@ def sleep_with_wakeup_events(loop_delay, wakeup_period = 5, options = {})
 
             if options[:clone_target_change]
               clone_target = redis.hget('vmpooler__config__clone_target}', options[:poolname])
-              if clone_target
-                break unless clone_target == initial_clone_target
-              end
+              break if clone_target && !clone_target == initial_clone_target
```
Comment (sbeaulie) [inline lib/vmpooler/pool_manager.rb]: unless clone_target is a Boolean, this is not the same. ! on a string returns false.
```suggestion
              break if clone_target && clone_target != initial_clone_target
```

**Review by sbeaulie — CHANGES_REQUESTED** [review]

File: lib/vmpooler/providers/vsphere.rb [hunk 2]
```diff
@@ -418,15 +416,15 @@ def set_network_device(datacenter_name, template_vm_network_device, network_name
           # Determine network device type
           # All possible device type options here: https://vdc-download.vmware.com/vmwb-repository/dcr-public/98d63b35-d822-47fe-a87a-ddefd469df06/2e3c7b58-f2bd-486e-8bb1-a75eb0640bee/doc/vim.vm.device.VirtualEthernetCard.html
           network_device =
-            if template_vm_network_device.is_a? RbVmomi::VIM::VirtualVmxnet2
```
Comment (sbeaulie) [inline lib/vmpooler/providers/vsphere.rb:421]: we need to check if there are subclasses that we also want to accept here? for example if `RbVmomi::VIM::VirtualE1000` has a subclass, `is_a?(RbVmomi::VIM::VirtualE1000)` would return `true` for the subclass too

File: lib/vmpooler/providers/vsphere.rb [hunk 3]
```diff
@@ -49,9 +51,7 @@ def name
         end
 
         def folder_configured?(folder_title, base_folder, configured_folders, whitelist)
-          if whitelist
-            return true if whitelist.include?(folder_title)
-          end
+          return true if whitelist&.include?(folder_title)
```
Comment (sbeaulie) [inline lib/vmpooler/providers/vsphere.rb:54]: what kind of magic is that!

Comment (rooneyshuman) [reply lib/vmpooler/providers/vsphere.rb:54]: the safe navigation operator is probably the coolest thing I saw from this rubocop change

**Review by sbeaulie — APPROVED** [review]

File: lib/vmpooler/providers/vsphere.rb (hunk 2 above)
Comment (rooneyshuman) [reply lib/vmpooler/providers/vsphere.rb:421]: Those network device classes don't have any subclasses. They are the exact class for the virtual network device objects

File: lib/vmpooler/providers/vsphere.rb (hunk 3 above)
Comment (jcoconnor) [reply lib/vmpooler/providers/vsphere.rb:54]: That does look rather cool.

**Review by jcoconnor — APPROVED** [review]

**Pull request merged**